src/utils/generic.rs

```diff
@@ -21,43 +24,56 @@ pub fn nonce(seed_nonce: &Scalar, label: &str, index: Option<usize>) -> Result<S
     let encoded_label = label.as_bytes();
     if encoded_label.len() > 16 {
         // See https://www.blake2.net/blake2.pdf section 2.8
-        return Err(ProofError::InvalidLength("nonce label".to_string()));
-    };
-    let hasher = if let Some(salt) = index {
+        Err(ProofError::InvalidLength("nonce label".to_string()))
+    } else if let Some(salt) = index {
```

Comment:
I prefer the old code here. 
We introduce more nesting by removing the return

Reply:
This is the way @DenisKolodin proposed to get rid of additional `return` statements.

Reply:
Reverted